gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/web/WebExtension.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020-2021 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.compose.web
+
+import org.gradle.api.Project
+import org.gradle.api.plugins.ExtensionAware
+import org.jetbrains.compose.internal.mppExt
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+import org.jetbrains.kotlin.gradle.targets.js.ir.KotlinJsIrTarget
+
+abstract class WebExtension : ExtensionAware {
+    private var requestedTargets: Set<KotlinJsIrTarget>? = null
+    private var targetsToConfigure: Set<KotlinJsIrTarget>? = null
+
+    // public api
+    @Suppress("unused")
+    fun targets(vararg targets: KotlinTarget) {
+        check(requestedTargets == null) {
+            "compose.web.targets() was already set!"
+        }
+
+        val jsIrTargets = linkedSetOf<KotlinJsIrTarget>()
+        for (target in targets) {
+            check(target is KotlinJsIrTarget) {
+                """|'${target.name}' is not a JS(IR) target:
```

Comment:
So we really cannot make configuration more automated?

Reply:
Note that it's not required, when there is one JS IR target.

As for our options:
* We cannot change a type of target via DSL, after a target is defined.  
* We can probably force IR compiler for all JS targets in a subproject by setting `kotlin.js.compiler=ir` for that particular subproject. A downside is that users will need another flag to override that behaviour.
* We can perform configuration for any defined JS IR target, even if there is more than one.
* We can also add an extension function, which would perform the necessary configuration, e.g.
```
kotlin {
     composeWeb() // or jsCompose or jsComposeWeb
     composeWeb("customTargetName")
}
```

Reply:
OK, let's keep it this way.